boa/contracts/vyper/vyper_contract.py

```diff
@@ -518,12 +518,14 @@ def __init__(
         created_from: Address = None,
         filename: str = None,
         gas=None,
+        nickname=None,
     ):
         super().__init__(compiler_data, env, filename)
 
         self.created_from = created_from
         self._computation = None
         self._source_map = None
+        self.nickname = nickname
```

Comment:
i think we should pass contract_name to the super ctor instead of nickname

Reply:
As in this? 

```python
self.nickname = contract_name if nickname is None else nicknick
```

Reply:
We don't need a new nickname field. We can just use contract_name